strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/PythonWheelArtifactGenerator.java

```diff
@@ -30,24 +36,61 @@
                                                    "\n" +
                                                    "Strongbox wheel package for test";
 
-    private String basedir;
+    private Path basedir;
+    
+    public PythonWheelArtifactGenerator(Path basedir)
+    {
+        this.basedir = basedir;
+    }
 
     public PythonWheelArtifactGenerator(String basedir)
     {
-        this.basedir = basedir;
+        this.basedir = Paths.get(basedir);
+    }
+
+    @Override
+    public Path generateArtifact(String id,
+                                 String version,
+                                 int size)
+            throws IOException
+    {
+        PypiWheelArtifactCoordinates coordinates = new PypiWheelArtifactCoordinates(id,
+                                                                                    version,
+                                                                                    null,
+                                                                                    "py3",
+                                                                                    "none",
+                                                                                    "any");
+        return generateArtifact(coordinates);
     }
 
-    public void generateWheelPackage(PypiWheelArtifactCoordinates coordinates)
+    @Override
+    public Path generateArtifact(URI uri,
+                                 int size)
             throws IOException
     {
-        String packagePath = String.format("%s/%s-%s-py2-none-any.whl", basedir, coordinates.getId(),
-                                           coordinates.getVersion());
+        PypiWheelArtifactCoordinates coordinates = PypiWheelArtifactCoordinates.parse(FilenameUtils.getName(uri.toString()));
+        return generateArtifact(coordinates);
+    }
 
-        ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(packagePath));
+    public Path generateArtifact(PypiWheelArtifactCoordinates coordinates)
+            throws IOException
+    {
+        String packagePath = coordinates.getId() + "-" +
```

Comment:
there is `coordinates.toPath()` which should do the same. I think we better use it here.